### PR TITLE
Add Empirical Software Engineering

### PR DESCRIPTION
**Pull request by dspinellis**
**[https://github.com/dspinellis/awesome-msr](https://github.com/dspinellis/awesome-msr)**

**Data sets associated with empirical software engineering research that is conducted by mining software repositories and by using search-based software engineering methods**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title.<br>For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with these requirements:

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of sub-lists, preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

**Comment by legodude17** [comment]
Suggestions:
 * Provide a description in this list's entry
 * Fix the link in this PR
 * Add a License in a LICENSE file
 * Remove `A /An` from descriptions

**Comment by dspinellis** [comment]
* [ ] Provide a description in this list's entry
* [ ] Fix the link in this PR
* [ ] Add a License in a LICENSE file
* [ ] Remove A /An from descriptions

**Comment by legodude17** [comment]
Sorry, missed that (somehow 🤷‍♂️ ).

**Comment by dspinellis** [comment]
No problem! Then all issues are addressed.

**Comment by sindresorhus** [comment]
Hey, sorry for the lack of response, please see #1363.

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/dspinellis/awesome-msr
✖ Linting

  README.md:24:5
  ✖   1:16   Invalid badge source                                                      remark-lint:awesome/badge
  ✖  15:1    ToC item "Data Sets" does not match corresponding heading "Repositories"  remark-lint:awesome/toc
  ✖  20:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  20:45   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  21:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  21:76   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  22:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  22:56   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  23:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  23:31   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  24:3    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  24:5    Don’t pad link with inner spaces                                          remark-lint:no-inline-padding
  ✖  25:3    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  26:3    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  30:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  30:61   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  31:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  31:39   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  32:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  32:60   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  33:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  33:59   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  34:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  34:64   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  35:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  35:50   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  36:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  37:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  37:103  List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  38:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  38:120  List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  39:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  39:37   Text "Github" should be written as "GitHub"                               remark-lint:awesome/spell-check
  ✖  39:37   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  40:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  40:84   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  41:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  41:38   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  42:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  42:120  List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  43:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  43:62   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  44:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  44:45   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  45:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  46:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  47:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  47:125  List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  48:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  48:48   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  49:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  50:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  51:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  51:83   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  52:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  52:42   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  53:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  53:66   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  57:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  57:36   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  58:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  58:43   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  59:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  59:47   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  60:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  60:47   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  61:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  61:67   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  63:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  63:211  List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  64:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  64:53   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  65:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  65:52   List item description must end with proper punctuation                    remark-lint:awesome/list-item
  ✖  66:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style
  ✖  67:1    Marker style should be -                                                  remark-lint:unordered-list-marker-style

  76 errors
```

https://github.com/sindresorhus/awesome-lint

File: readme.md [hunk 1]
```diff
@@ -283,6 +283,7 @@
 - [Functional Programming](https://github.com/lucasviola/awesome-functional-programming)
 - [Static Analysis & Code Quality](https://github.com/mre/awesome-static-analysis)
 - [Information Retrieval](https://github.com/harpribot/awesome-information-retrieval) - Learn to develop your own search engine.
+- [Mining Software Repositories and Search-Based Software Engineering](https://github.com/dspinellis/awesome-msr) - Data sets for conducting empirical software engineering research by mining software repositories or by using search-based software engineering methods
```
Comment (sindresorhus) [inline readme.md]: Both the list title and description are too verbose.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by dspinellis** [comment]
Thank you for coming back to this. I fixed the merge conflict and shortened the title and the description.

File: readme.md [hunk 2]
```diff
@@ -287,6 +287,7 @@
 - [Deep Vision](https://github.com/kjw0612/awesome-deep-vision#readme)
 - [Open Source Society University](https://github.com/ossu/computer-science#readme)
 - [Functional Programming](https://github.com/lucasviola/awesome-functional-programming#readme)
+- [Empirical Software Engineering](https://github.com/dspinellis/awesome-msr#readme) - Data sets and tools for conducting empirical software engineering research.
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should:
> * Include a short description about the project/theme of the list. **It should not describe the list itself.**

It should describe **what** `Empirical Software Engineering` is.

**Comment by sindresorhus** [comment]
Your list heading is completely different from the title you've used here.

**Comment by dspinellis** [comment]
Thank you for the swift response. I updated the list's heading to match the given title.

File: readme.md (hunk 2 above)
Comment (dspinellis) [reply readme.md]: I changed the description to describe the content, rather than the list.

**Comment by dspinellis** [comment]
> 
> 
> ```
> ❯ awesome-lint https://github.com/dspinellis/awesome-msr
> ✖ Linting
> 
[...]
> https://github.com/sindresorhus/awesome-lint

Thank you for the pointer. Fixed and lint CI added to the project.

**Comment by sindresorhus** [comment]
The list heading should be `Awesome Empirical Software Engineering`, not `Awesome Empirical Software Engineering Resources`.

---	

You should not use my email in https://github.com/dspinellis/awesome-msr/blob/master/code-of-conduct.md

**Comment by dspinellis** [comment]
Good points. I corrected both identified issues.

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1126388832805330945

**Pull request merged**